src/components/CreateAccount.js

```diff
@@ -56,59 +71,230 @@ class CreateAccount extends Component {
         });
     }
 
+    /**
+     * Signs the user in after account creation.
+     */
+    signIn() {
+        this.props.client.authenticate(this.state.username, this.state.password)
+            .then(response => {
+                if (response.ok) {
+                    // Notifiy the parent component that account creation and
+                    // sign-in are complete
+                    this.props.onSignIn(true, this.state.username);
+                    this.setState({ isSignedIn: true });
+                } else {
+                    // Trigger a redirect to the SignIn page if the account
+                    // creation was successful but the sign-in failed
+                    this.setState({ goToSignIn: true });
+                }
+            });
+    }
+
     /**
      * Updates the component state when the user types in an input field.
      * @param {Event} event
      */
-    handleInputChange(event) {
+    handleInput(event) {
+        // Make sure the value matches the requirements for its field
+        const value = event.target.value;
+        switch (event.target.name) {
+        case 'username':
+            this.setState({
+                'usernameIsValid': this.validateUsername(value)
+            });
+            break;
+        case 'email':
+            this.setState({
+                'emailIsValid': this.validateEmail(value)
+            });
+            break;
+        case 'password':
+            this.setState({
+                'passwordIsValid': this.validatePassword(value)
+            });
+            break;
+        }
+
         this.setState({
-            [event.target.name]: event.target.value
+            [event.target.name]: value,
         });
     }
 
+    /**
+     * Returns true if the username is valid and false otherwise
+     * @param {String} username
+     */
+    validateUsername(username) {
```

Comment:
maybe in the future, we can replace all this logic with some package that does it for us